src/Service/Resolvers/SqlQueryEngine.cs

```diff
@@ -213,6 +215,40 @@ private OkObjectResult FormatFindResult(JsonDocument jsonDoc, FindRequestContext
             // nextLink is the URL needed to get the next page of records using the same query options
             // with $after base64 encoded for opaqueness
             string path = UriHelper.GetEncodedUrl(_httpContextAccessor.HttpContext!.Request).Split('?')[0];
+
+            RuntimeConfig runtimeConfig = _runtimeConfigProvider.GetConfig();
+            // If the base route is not empty, we need to insert it into the URI before the rest path.
+            string? baseRoute = runtimeConfig.Runtime.Rest.BaseRoute;
+            if (!string.IsNullOrEmpty(baseRoute))
+            {
+                _sqlMetadataProvider.TryGetEntityPathFromName(context.EntityName, out string? pathNameForEntity);
+
+                // Path is of the form ....restPath/pathNameForEntity. We want to insert the base route before the restPath,
+                // hence we remove pathNameForEntity and restPath from the path and append them back after appending the base route to the path.
+                // Initialize lastIdxOfPathToBeIncluded by removing the pathNameForEntity from the path.
+                int lastIdxOfPathToBeIncluded = path.Length - pathNameForEntity!.Length - 1;
```

Comment:
Performing this for each request seems computationally expensive. Can we optimize in any way?

Reply:
I'm afraid not, will give it some more thought.

Reply:
Due to the dynamic nature of uri, I don't think we can kind of precalculate it instead of doing it on a per request basis.

Reply:
Incorporated https://github.com/Azure/data-api-builder/pull/1506#discussion_r1254974637